### PR TITLE
Copy logging sidecar entrypoint to 2.2.3 image

### PR DESCRIPTION
**Pull request by thmzlt**
**What this PR does / why we need it**: Copy logging sidecar entrypoint to 2.2.3 image

**Special notes for your reviewer**: These are the changes suggested by @rob-1126 in his implementation of the sidecar logging feature. I copied them verbatim.

Related to astronomer/issues#3563.

### CODE REVIEW

File: 2.2.3/bullseye/include/entrypoint [hunk 1]
```diff
@@ -50,3 +100,9 @@ fi
 
 # Run the original command
 exec "$@"
+
+
+# If we are using a sidecar logging keepalive socket hang up
+if [[ ! -z ${SIDECAR_LOGGING_KEEPALIVE_SOCKET+x} ]] && [[ "$COMPONENT" =~ ^(scheduler|webserver|worker)$ ]]; then
+  echo "goodbye" >> $SIDECAR_LOGGING_KEEPALIVE_SOCKET
+fi
```
Comment (danielhoherd) [inline 2.2.3/bullseye/include/entrypoint:108]: This appears to be not run any time there is any content in `"$@"`. Tested via:

```
$ cat foo.sh
#!/usr/bin/env bash
echo hi
exec "$@"
echo bye
$ ./foo.sh
hi
bye
$ ./foo.sh whoami
hi
danielh
```

File: 2.2.3/bullseye/include/entrypoint [hunk 2]
```diff
@@ -1,5 +1,55 @@
...
+    echo "COMPONENT must be set when using sidecar logging."
+    exit 2;
+  fi
+  UNSTRUCTURED_LOGGING_SOCKET=/dev/tcp/${UNSTRUCTURED_LOGGING_HOST}/${UNSTRUCTURED_LOGGING_PORT}
+  echo "Connecting to tcp logging daemon on ${UNSTRUCTURED_LOGGING_SOCKET}"
+  while true; do
+    set +e
+    # create a pointer to the origianl stdout and stderr
+    exec 3>&1 4>&2 2>&4
+    # clobber old stdout and stderr
```
Comment (danielhoherd) [inline 2.2.3/bullseye/include/entrypoint:37]: Is this supposed to have a line with some redirects below it?

**Comment by kaxil** [comment]
Don't we need it for other images as well, or are we intentionally targeting 2.2.3 and above?

**Comment by thmzlt** [comment]
> Don't we need it for other images as well, or are we intentionally targeting 2.2.3 and above?

@kaxil, that question came up on our conversation yesterday. Rob initially added support for logging sidecar on 2.1.3 and I did 2.2.3 because it was the latest version. 

@rob-1126 can you check Daniel's comments when you have a chance?

File: 2.2.3/bullseye/include/entrypoint (hunk 2 above)
Comment (rob-1126) [reply 2.2.3/bullseye/include/entrypoint:39]: I think the connection happens on 36. If it was successful it lets the redirect stand.

**Comment by kaxil** [comment]
> @kaxil, that question came up on our conversation yesterday. Rob initially added support for logging sidecar on 2.1.3 and I did 2.2.3 because it was the latest version.

I don't see it in 2.1.3 though

**Comment by danielhoherd** [comment]
@kaxil Thomaz was waiting until everything was approved before copying it to the other supported versions.

File: 2.2.3/bullseye/include/entrypoint [hunk 3]
```diff
@@ -1,5 +1,58 @@
 #!/usr/bin/env bash
 set -e
+set -x
+
+
+# Optional sidecar logging keepalive support
+if [[ -n ${SIDECAR_LOGGING_KEEPALIVE_PORT+x} ]] && [[ "$COMPONENT" =~ ^(scheduler|webserver|worker)$ ]] && [[ $1 != "python" ]]; then
```
Comment (kaxil) [inline 2.2.3/bullseye/include/entrypoint:7]: Why +x here?

Comment (danielhoherd) [reply 2.2.3/bullseye/include/entrypoint:7]: That's an old suggested syntax that we should not be using. https://github.com/koalaman/shellcheck/wiki/SC2268

Comment (danielhoherd) [reply 2.2.3/bullseye/include/entrypoint:7]: wait, no. that's a variable substitution that says "If SIDECAR_LOGGING_KEEPALIVE_PORT is defined, print x." i'm not sure why we would use that here. in any case i will fix it.

File: 2.2.3/bullseye/include/entrypoint (hunk 2 above)
Comment (danielhoherd) [reply 2.2.3/bullseye/include/entrypoint:39]: What i mean is there's a comment that says "Clobber old stdout and stderr" but there is nothing below that comment.

**Comment by rob-1126** [comment]
Agreed. That comment doesn't seem to describe what's going on there at all.
I suggest snipping lines 33-45 and reimplementing a retry fail there that's
less terrible. That code is certainly terrible and entirely untested (my
doing). Lines 33-45 look to not serve their intended purpose at all
(definitely my fault). It looks like at some time in the past lines 46-51
probably resided between lines 38 and 39.

basically to do something in bash like:
while true:
  try:
     setup_redirect()
     break
  catch:
    sleep(.05)


#

File: 2.2.3/bullseye/include/entrypoint (hunk 1 above)
Comment (danielhoherd) [reply 2.2.3/bullseye/include/entrypoint]: I removed the `exec` so this should work now.

**Comment by jedcunningham** [comment]
(Closing this because it is stale)

**Comment by danielhoherd** [comment]
FTR this approach as abandoned, so this PR wasn't needed anymore.
